html/pfappserver/root/src/views/Configuration/roles/config.js

```diff
@@ -0,0 +1,69 @@
+import i18n from '@/utils/locale'
+import { pfSearchConditionType as conditionType } from '@/globals/pfSearch'
+
+export const columns = [
+  {
+    key: 'id',
+    label: i18n.t('Name'),
+    required: true,
+    sortable: true,
+    visible: true,
+    searchable: true
+  },
+  {
+    key: 'notes',
+    label: i18n.t('Description'),
+    sortable: true,
+    visible: true,
+    searchable: true
+  },
+  {
+    key: 'max_nodes_per_pid',
+    label: i18n.t('Max nodes per user'),
+    sortable: true,
+    visible: true
+  },
+  {
+    key: 'buttons',
+    locked: true
+  },
+  {
+    key: 'children',
+    required: true
+  },
+  {
+    key: 'parent_id',
+    required: true
+  },
+  {
+    key: 'not_deletable',
+    required: true
+  }
+]
+
+export const fields = [
+  {
+    value: 'id',
+    text: i18n.t('Name'),
+    types: [conditionType.SUBSTRING]
+  },
+  {
+    value: 'notes',
+    text: i18n.t('Description'),
+    types: [conditionType.SUBSTRING]
+  }
+]
+
+export const reasons = {
```

Comment:
Add a comment on the origin of those key strings. `lib/pf/UnifiedApi/Controller/Config/Roles.pm`?